File: android_mobile/src/main/java/com/alexstyl/specialdates/events/namedays/NamedayDatabaseRefresher.java
package com.alexstyl.specialdates.events.namedays;

import com.alexstyl.specialdates.date.ContactEvent;
import com.alexstyl.specialdates.events.peopleevents.PeopleEventsPersister;
import com.alexstyl.specialdates.events.peopleevents.PeopleNamedaysCalculator;

import java.util.List;

public class NamedayDatabaseRefresher {

    private final NamedayUserSettings namedayUserSettings;
    private final PeopleEventsPersister perister;
    private final PeopleNamedaysCalculator calculator;

    public NamedayDatabaseRefresher(NamedayUserSettings namedayPreferences,
                                    PeopleEventsPersister databaseProvider,
                                    PeopleNamedaysCalculator calculator) {
        this.namedayUserSettings = namedayPreferences;
        this.perister = databaseProvider;
        this.calculator = calculator;
    }

    public void refreshNamedaysIfEnabled() {
        perister.deleteAllNamedays();
        if (namedayUserSettings.isEnabled()) {
            initialiseNamedays();
        }
    }

    private void initialiseNamedays() {
        List<ContactEvent> namedays = calculator.loadDeviceStaticNamedays();
        storeNamedaysToDisk(namedays);
    }

    private void storeNamedaysToDisk(List<ContactEvent> namedays) {
        perister.insertAnnualEvents(namedays);
    }

}

File: memento/src/main/java/com/alexstyl/specialdates/events/namedays/NamedayDatabaseRefresher.kt
package com.alexstyl.specialdates.events.namedays

import com.alexstyl.specialdates.date.ContactEvent
import com.alexstyl.specialdates.events.peopleevents.PeopleEventsPersister
import com.alexstyl.specialdates.events.peopleevents.PeopleDynamicNamedaysProvider

class NamedayDatabaseRefresher(private val namedayUserSettings: NamedayUserSettings,
                               private val perister: PeopleEventsPersister,
                               private val provider: PeopleDynamicNamedaysProvider) {

    fun refreshNamedaysIfEnabled() {
        perister.deleteAllNamedays()
        if (namedayUserSettings.isEnabled) {
            initialiseNamedays()
        }
    }

    private fun initialiseNamedays() {
        val namedays = provider.loadAllStaticNamedays()
        storeNamedaysToDisk(namedays)
    }

    private fun storeNamedaysToDisk(namedays: List<ContactEvent>) {
        perister.insertAnnualEvents(namedays)
    }

}
